### ISSUE TITLE
kubelet panic on new node when Node resource already exists

### ISSUE THREAD

**Issue by neiljerram** [kubernetes/kubernetes#100862]
#### What happened:

kubelet service panicked and therefore failed to register a newly booted node with the cluster:
```
...
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: I0406 05:04:15.268794   26208 plugin_manager.go:114] Starting Kubelet Plugin Manager
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: I0406 05:04:15.271257   26208 container_manager_linux.go:988] Found 211 PIDs in root, 211 of them are not to be moved
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: I0406 05:04:15.273822   26208 kubelet_node_status.go:108] Node master0.ocp4.contoso.com was previously registered
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: I0406 05:04:15.273835   26208 kubelet_node_status.go:249] Controller attach-detach setting changed to true; updating existing Node
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: E0406 05:04:15.273916   26208 runtime.go:78] Observed a panic: "assignment to entry in nil map" (assignment to entry in nil map)
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: goroutine 304 [running]:
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x3d74e20, 0x4b5f230)
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]:         /builddir/build/BUILD/openshift-git-94035.af27205/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa6
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]:         /builddir/build/BUILD/openshift-git-94035.af27205/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x89
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: panic(0x3d74e20, 0x4b5f230)
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]:         /usr/lib/golang/src/runtime/panic.go:969 +0x175
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: k8s.io/kubernetes/pkg/kubelet.(*Kubelet).reconcileHugePageResource(0xc00055a000, 0xc000560000, 0xc001402000, 0x0)
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]:         /builddir/build/BUILD/openshift-git-94035.af27205/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet_node_status.go:146 +0x47b
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: k8s.io/kubernetes/pkg/kubelet.(*Kubelet).tryRegisterWithAPIServer(0xc00055a000, 0xc000560000, 0x0)
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]:         /builddir/build/BUILD/openshift-git-94035.af27205/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet_node_status.go:116 +0x48b
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: k8s.io/kubernetes/pkg/kubelet.(*Kubelet).registerWithAPIServer(0xc00055a000)
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]:         /builddir/build/BUILD/openshift-git-94035.af27205/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet_node_status.go:71 +0x205
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: k8s.io/kubernetes/pkg/kubelet.(*Kubelet).syncNodeStatus(0xc00055a000)
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]:         /builddir/build/BUILD/openshift-git-94035.af27205/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet_node_status.go:426 +0x14d
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000ce1f70)
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]:         /builddir/build/BUILD/openshift-git-94035.af27205/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000ce1f70, 0x4c19100, 0xc0006e5170, 0x1, 0xc0000ac0c0)
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]:         /builddir/build/BUILD/openshift-git-94035.af27205/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xad
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000ce1f70, 0x2540be400, 0x0, 0x1, 0xc0000ac0c0)
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]:         /builddir/build/BUILD/openshift-git-94035.af27205/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc000ce1f70, 0x2540be400, 0xc0000ac0c0)
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]:         /builddir/build/BUILD/openshift-git-94035.af27205/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]: created by k8s.io/kubernetes/pkg/kubelet.(*Kubelet).Run
Apr 06 05:04:15 master0.ocp4.contoso.com hyperkube[26208]:         /builddir/build/BUILD/openshift-git-94035.af27205/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1344 +0x398
```

#### What you expected to happen:

kubelet registered the new node with the cluster.

#### How to reproduce it (as minimally and precisely as possible):

Before booting up the new node (or before starting the kubelet service on it), create a matching Node resource like this:
```
apiVersion: v1
kind: Node
metadata:
  name: master0.ocp4.contoso.com
  labels:
    rack: ra
    node-role.kubernetes.io/master: ""
  annotations:
    projectcalico.org/ASNumber: "65001"
```

Then boot the new node (or start the kubelet service on it).

#### Anything else we need to know?:

I'm doing this because I want the node to be labelled (with `rack: ra`) and annotated (with `projectcalico.org/ASNumber: "65001"`) very early in its lifetime.  This approach of creating a Node resource in advance worked in Kubernetes 1.18, but no longer works in 1.19.  The kubelet code does have handling for an `existingNode`, so that suggests the approach should work.

#### Environment:
- Kubernetes version (use `kubectl version`): 1.19 I believe.
- Cloud provider or hardware configuration: OpenShift 4.6
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug): Calico, but this problem occurs on a new node before any Calico code starts running.
- Others:

**Comment by dims** [comment]
Hey @neiljerram, we are unable to support k8s distros. can you please file this in openshift github repo?

/close

**Comment by pacoxu** [comment]
dup with #95188  and is fixed in #96297 for v1.19.6+.

**Comment by neiljerram** [comment]
Awesome, thank you @pacoxu !
